### PR TITLE
Updating to use effective viewport feature when running on OS above RS5

### PR DESCRIPTION
**Pull request by ranjeshj**
Switch to using effective viewport when running on OS >= RS5. When running below RS5 we will use the downlevel support. This will allow using MUX.ItemsRepeater with WUX.ScrollViewer in RS5+ and have virtualization still work.

Internal Issue: https://microsoft.visualstudio.com/OS/_queries/edit/19562207/?triage=true
Test Run: https://microsoft.visualstudio.com/WinUI/WinUI%20Team/_build/results?buildId=13319592

### CODE REVIEW

File: dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs [hunk 1]
```diff
@@ -58,6 +56,12 @@ public class EffectiveViewportTests : TestsBase
         [TestMethod]
         public void ValidateNoScrollingSurfaceScenario()
         {
+            if (!PlatformConfiguration.IsOsVersionGreaterThanOrEqual(OSVersion.Redstone5))
+            {
+                Log.Warning("Skipping since version is less than RS5");
```
Comment (chrisglein) [inline dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs]: Be more crisp about what RS5 feature is necessary for this?

File: dev/Repeater/APITests/ViewportTests.cs [hunk 2]
```diff
@@ -348,6 +348,13 @@ public void ValidateSupportForScrollerConfigurationChanges()
                 return;
             }
 
+            // Post Rs5 configuration changes will not be raised.
```
Comment (RBrid) [inline dev/Repeater/APITests/ViewportTests.cs]: Shouldn't "Rs5" say "RS4" here?

File: dev/Repeater/APITests/ViewportTests.cs [hunk 3]
```diff
@@ -348,6 +348,13 @@ public void ValidateSupportForScrollerConfigurationChanges()
...
+            if (PlatformConfiguration.IsOsVersionGreaterThanOrEqual(OSVersion.Redstone5))
+            {
+                Log.Warning("Skipping ValidateSupportForScrollerConfigurationChanges");
```
Comment (chrisglein) [inline dev/Repeater/APITests/ViewportTests.cs]: In the logs this will be indistinguishable from the above Warning. It'd be better if the log message contained more identifiable information.

File: dev/Repeater/TestUI/RepeaterTestUIPage.xaml [hunk 4]
```diff
@@ -41,8 +41,8 @@
                 <TextBlock>MinColumnSpacing</TextBlock>
                 <TextBox x:Name="lineSpacing" Text="10"></TextBox>
                 <TextBlock>LineAlignment</TextBlock>
-                <TextBlock>(Near/Middle/Far/SpaceEqually)</TextBlock>
-                <TextBox x:Name="lineAlingment" Text="Near"></TextBox>
+                <TextBlock>(Start/Center/End/SpaceAround/SpaceBetween/SpaceEvenly)</TextBlock>
```
Comment (chrisglein) [inline dev/Repeater/TestUI/RepeaterTestUIPage.xaml]: Is this related to the rest of this change? From the description it sounded like it was only about fixing the downlevel and OS build stuff.

File: dev/Scroller/Scroller.cpp [hunk 5]
```diff
@@ -808,10 +808,11 @@ winrt::Size Scroller::ArrangeOverride(winrt::Size const& finalSize)
         viewport.Width          /*viewportWidth*/,
         viewport.Height         /*viewportHeight*/);
 
-#ifndef USE_EFFECTIVE_VIEWPORT_AND_ANCHORING_FROM_PLATFORM
-    ClearAnchorCandidates();
-    RaisePostArrange();
-#endif
+    if (!SharedHelpers::IsRS5OrHigher())
```
Comment (chrisglein) [inline dev/Scroller/Scroller.cpp]: It's not obvious from these blocks _why_ they need to be making version checks

File: dev/Repeater/TestUI/RepeaterTestUIPage.xaml (hunk 4 above)
Comment (ranjeshj) [reply dev/Repeater/TestUI/RepeaterTestUIPage.xaml]: Not related but this was causing a crash in the test app due to a rename that was missed in a previous change.

File: dev/Scroller/Scroller.cpp (hunk 5 above)
Comment (ranjeshj) [reply dev/Scroller/Scroller.cpp]: Will add some comments.

File: dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs [hunk 6]
```diff
@@ -48,80 +48,73 @@
...
+                        Log.Comment("ViewChanged " + scrollViewer.HorizontalOffset + ":" + scrollViewer.VerticalOffset);
+                        viewChanged.Set();
+                    }
+                };
+
                 Content = scrollViewer;
-                Content.UpdateLayout();
+            });
+
+            Verify.IsTrue(layoutMeasured.WaitOne(), "Did not recieve measure on layout");
```
Comment (RBrid) [inline dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs]: recieve --> receive everywhere.

File: dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs [hunk 7]
```diff
@@ -154,6 +156,12 @@ public void ValidateBasicScrollViewerScenario()
         [TestMethod]
         public void ValidateOneScrollViewerScenario()
         {
+            if (!PlatformConfiguration.IsOsVersionGreaterThanOrEqual(OSVersion.Redstone5))
+            {
+                Log.Warning("Skipping since version is less than RS5 since effective viewport is not available below RS5");
```
Comment (RBrid) [inline dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs]: since --> and.

File: dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs [hunk 8]
```diff
@@ -48,80 +48,73 @@
...
-                // 32 pixel title bar
-                Verify.AreEqual(-32, realizationRects[1].Y);
-                // Width/Height depends on the window size, so just
-                // validating something reasonable here to avoid flakiness.
-                Verify.IsLessThan(500.0, realizationRects[1].Width);
-                Verify.IsLessThan(500.0, realizationRects[1].Height);
-                realizationRects.Clear();
-            });
-        }
+                Log.Warning("Skipping since version is less than RS5 since effective viewport is not available below RS5");
```
Comment (RBrid) [inline dev/Repeater/APITests/EffectiveViewportScrollViewerTests.cs]: since --> and

File: dev/Repeater/ViewportManagerDownlevel.h [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    void RegisterCacheBuildWork();
+    void TryInvalidateMeasure();
+    winrt::IRepeaterScrollingSurface GetOuterScroller() const;
+
+    winrt::hstring GetLayoutId();
+
+    ItemsRepeater* m_owner{ nullptr };
+
+    // List of parent scrollers.
+    // The list stop when we reach the root scroller OR when both m_horizontalScroller
```
Comment (RBrid) [inline dev/Repeater/ViewportManagerDownlevel.h]: stop --> stops

File: dev/Scroller/APITests/ScrollerViewChangeTests.cs [hunk 10]
```diff
@@ -1604,9 +1605,16 @@ private void ChangeZoomFactorBeforeLoading(bool animate)
         private void WaitForEvent(string logComment, EventWaitHandle eventWaitHandle)
         {
             Log.Comment(logComment);
-            if (!eventWaitHandle.WaitOne(TimeSpan.FromMilliseconds(c_MaxWaitDuration)))
+            if(Debugger.IsAttached)
```
Comment (RBrid) [inline dev/Scroller/APITests/ScrollerViewChangeTests.cs]: missing space.
